[revealjs] Scroll View documentation (#1387)
quarto-dev/quarto-web: docs/presentations/revealjs/presenting.qmd

This article covers the mechanics of presenting slides with Reveal. Basic navigation is done using the following keyboard shortcuts:

+----------------------------+--------------------------------------------+
| Action                     | Keys                                       |
+============================+============================================+
| Next slide                 | <kbd>→</kbd> <kbd>SPACE</kbd> <kbd>N</kbd> |
+----------------------------+--------------------------------------------+
| Previous slide             | <kbd>←</kbd> <kbd>P</kbd>                  |
+----------------------------+--------------------------------------------+
| Navigate without fragments | <kbd>Alt →</kbd> <kbd>Alt ←</kbd>          |
+----------------------------+--------------------------------------------+
| Jump to first/last slide   | <kbd>Shift →</kbd> <kbd>Shift ←</kbd>      |

> +----------------------------+--------------------------------------------+

Reviewer: While we are in this file, is it worth changing these to use our `kbd` shortcode?

Author: It seems kbd shortcode in table is not working well... 

````markdown
+----------------------------+-------------------------------------------------+
| Action                     | Keys                                            |
+============================+=================================================+
| Next slide                 | {{< kbd → >}} {{< kbd SPACE >}} {{< kbd N >}}   |
+----------------------------+-------------------------------------------------+
| Previous slide             | {{< kbd ← >}} {{< kbd P >}}                     |
+----------------------------+-------------------------------------------------+
| Navigate without fragments | {{< kbd Alt + → >}} {{< kbd Alt + ← >}}         |
+----------------------------+-------------------------------------------------+
| Jump to first/last slide   | {{< kbd Shift + → >}} {{< kbd Shift + ← >}}     |
+----------------------------+-------------------------------------------------+
````
It seems like a good table, but it gives me this 

![image](https://github.com/user-attachments/assets/155b7062-c39d-4d4d-b88a-4ab834e3bc9f)

And I can't make it works

Note that the `secret` value will be included in only the speaker version of the presentation.

## Scroll View

{{< include /docs/prerelease/1.6/_pre-release-feature.qmd >}}

> Reveal presentations be presented in [Scroll View](https://revealjs.com/scroll-view/) with vertically scrolled navigation instead of Slides. This is a new view mode introduced in Revealjs 5. It is is always available in all presentation as an alternative view mode (like [Print view](#print-to-pdf)), and it will be used a default view on small screen (e.g. viewing a presentations on mobile).

Reviewer: ```suggestion
Reveal presentations can be presented in [Scroll View](https://revealjs.com/scroll-view/) with vertically scrolled navigation instead of Slides. This is a new view mode introduced in Revealjs 5. It is always available in presentations as an alternative view mode (like [Print view](#print-to-pdf)), and it will be used as the default view on small screens (e.g. viewing a presentation on a mobile).
```
Why the capitalization on "Slides"? Do we use that consistently?

Author: Probably no reason... Thanks !

### Entering Scroll View {#scroll-view-toggle}

> To enter scroll view , do one the following:

Reviewer: ```suggestion
To enter Scroll View , do one the following:
```
We need to use capitalization consistently for "Scroll View" and possibly make it consistent with "Print view". I'm not sure exactly which style of capitalization makes sense. I see the Revealjs docs use "scroll view" (all lower case) in prose.

Author: I am fine with using `scroll view` lowercase. I am not that good on knowing this kind of things. I guess I did to make `Scroll View` kind a of thing a name of feature and not just tow word term. But anyway I am fine.

-   Toggle into Scroll View using the {{< kbd R >}} key

-   Toggle into Scroll View using the [Navigation Menu](#navigation-menu)

> -   Toggle into Scroll View by adding `?view=scroll` to the url, e.g. `https://quarto.org/docs/presentations/revealjs/demo/?view=scroll`

Author: Currently one need to go to the prerelease url to see a demo 
https://prerelease.quarto.org/docs/presentations/revealjs/demo/?view=scroll

Reviewer: Even with the prerelease URL I can't get the other two options (navigation menu, or `R`) to work.

Author: Oh yeah I know. I need to build a pre-release version with my scroll view change, and that the website pre-release is built with it. I am onto it.

Author: And in fact it seems I have an issue... I'll fix it. Thanks for catching ! I was sure I tested 🤔

Author: Should be solved now following #1389

You can easily toggle out of Scroll View by the same action.

> ### Default view mode

Reviewer: ```suggestion
### Default View Mode
```

> Scroll View can be made as default mode for the presentation, using default configuration. This can be configure using `scroll-view` configuration.

Reviewer: ```suggestion
Scroll View can be set as the default mode for the presentation using the `scroll-view` option:
```

``` yaml
format:
  revealjs:
    scroll-view: true
```

### Scroll View Options

> Scroll View in Reveal can be configured using the following option under `scroll-view`.

Reviewer: ```suggestion
Scroll View can be configured using the following options under `scroll-view`:
```

+--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Option             | Description                                                                                                                                                                                                                                  |
+====================+==============================================================================================================================================================================================================================================+

> | `progress`         | Show the scrollbar while scrolling and hide while idle (default `auto`). Set to `true` to always show, `false` to always hide.                                                                                                               |

Reviewer: This applies to all of these options: I'd reorder the first sentence so that the value is first, then a description of it. E.g.
```suggestion
| `progress`         |  `auto`: Show the scrollbar while scrolling and hide while idle (default). Set to `true` to always show, `false` to always hide.                                                                                                               |
```

| `snap`             | When scrolling, it will automatically snap to the closest slide (`mandatory`). Only snap when close to the top of a slide using `proximity`. Disable snapping altogether by setting to `false`.                                              |
+--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| `layout`           | By default each slide will be sized to be as tall as the viewport (default `full)`. If you prefer a more dense layout with multiple slides visible in parallel, set to `compact`.                                                            |
+--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> | `activation-width` | Control scroll view activation width. The scroll view is automatically unable when the viewport reaches mobile widths. Set to `0` to disable automatic scroll view or to another value to automatically trigger on bigger or smaller widths. |

Reviewer: ```suggestion
| `activation-width` | Control scroll view activation width. The scroll view is automatically activated when the viewport reaches mobile widths. Set to `0` to disable automatic scroll view or to another value to automatically trigger on bigger or smaller widths. |
```
What units is this? Pixels?

Author: Yes I think so. Default is 435 and I believe it is pixels I'll add it

|                    | scroll-view:                                                                                                                                                                                                                                 |
|                    |   activate: true                                                                                                                                                                                                                             |
|                    | ```                                                                                                                                                                                                                                          |
+--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> For example, the following specifies that we want a compact layout with proximity snap mode, and always showing the progress scrollbar while keeping activation to scroll view manual, even disabling on mobile view:

Reviewer: ```suggestion
For example, the following specifies that we want a compact layout, proximity snap mode, to always show the progress scrollbar, and to never automatically activate for small screens:
```
